### ISSUE TITLE
Custom toolbar buttons or triggers

### ISSUE THREAD

**Issue by AidanDaniel97** [apexcharts/apexcharts.js#299]
I am looking to be able to add custom buttons to the toolbar such as custom filters etc via the vue component or the ability to trigger SVG download as I need this ability. I would much prefer it to all be in one place apposed to having a separate filter

**Comment by junedchhipa** [comment]
What options would you need for custom buttons?

I can think of
```javascript
chart: {
  toolbar: {
    customIcons: [{
      html: '<i class="fa fa-angle-down"></i>',
      onClick: function(e, chartContext) { ... },
      appendTo: 'left' // left / top means the button will be appended to the left most or right most position
     }]
  }
}
```

If you have anything else in mind, please suggest

**Comment by AidanDaniel97** [comment]
Something along those lines would be great! Can't think of any other options that would be needed, at least not that I would need

**Comment by junedchhipa** [comment]
Ok, I will implement it in the next minor release

**Comment by darkoromanov** [comment]
+1 for this request, that would be really useful and powerful!

**Comment by darkoromanov** [comment]
I did a pull request with this feature https://github.com/apexcharts/apexcharts.js/pull/329

The new option would be like:

                 ```
customIcons: [{
                    icon: '<img src="/static/icons/chart-carpet.png" width="20">',
                    class: 'custom-icon',
                    index: 0,
                    click: function () {
                      console.log("ok")
                    }
                  }]
```

**Comment by junedchhipa** [comment]
Thanks @darkoromanov 
I modified the code to allow the index of those controls
The custom toolbar options would be like this.
```javascript
                  customIcons: [{
                    icon: '<img src="/static/icons/chart-carpet.png" width="20">',
                    index: 4,
                    class: 'custom-icon',
                    click: function () {
                      console.log("cliked")
                    }
                  }]
```

**Comment by darkoromanov** [comment]
Cool thanks!

**Comment by GonzalezAnguita** [comment]
Hi!
Is there any reason for not supporting JSX-style objects as icons?

```
import { FiDownload } from 'react-icons/fi';
...
tools: {
        download: <FiDownload />,
        selection: true,
        zoom: true,
        zoomin: true,
        zoomout: true,
        pan: false,
        reset: '',
        customIcons: [],
}
```

Would be awesome to have support for [React-Icons](https://react-icons.netlify.com/)

**Comment by pamelasouza** [comment]
> Obrigado @darkoromanov Modifiquei
> o código para permitir o índice desses controles
> As opções personalizadas da barra de ferramentas seriam assim.
> 
> ```js
>                   customIcons: [ { 
>                     icon : '<img src = "/ static / icons / chart-carpet.png" width = "20">' , 
>                     índice : 4 , 
>                     título : 'dica de ferramenta do ícone' , 
>                     classe : 'custom-icon ' , 
>                     clique em : function  ( )  { 
>                       console . log ( "clicado" ) 
>                     } 
>                   } ]
> ```

Please, how do I replace the icon with another svg? Or another local PNG icon?


This did not work:
```js
       zoom: `<img src={require('./icons-chart/teste.png')} width="20">`
```

```js
       zoom: '<img src={require(`./icons-chart/teste.png`)} width="20">',
```

```js
       zoom: '<img src="./icons-chart/teste.png" width="20">',
```

```js
       zoom: '<img src="./icons-chart/teste.svg" width="20">',
```

```js
       zoom: '<img src=`${varSVG}` width="20">',
```
I tried some other ways and none of them rendered.

To test, I added the two icons (svg and png) to a <img src> tag to make sure they are rendering outside the lib.

**Comment by KresimirJurkovic** [comment]
Hi @pamelasouza , have you maybe found a solution? My code, as below, is not working:
`toolbar: {
          download:
            '<img src="../../assets/img/dehaze-white.png" width="20" height="20"></img>',
          customIcons: [
            {
              icon:
                '<img src="../../assets/img/dehaze-white.png" width="20" height="20"></img>',
              index: 4,
              title: "tooltipp",
              class: "custom-icon",
              click: () => {
                console.log("clicked");
              }
            }
          ]
        }`

**Comment by KresimirJurkovic** [comment]
> Thanks @darkoromanov
> I modified the code to allow the index of those controls
> The custom toolbar options would be like this.
> 
> ```js
>                   customIcons: [{
>                     icon: '<img src="/static/icons/chart-carpet.png" width="20">',
>                     index: 4,
>                     title: 'tooltip of the icon',
>                     class: 'custom-icon',
>                     click: function () {
>                       console.log("cliked")
>                     }
>                   }]
> ```

Is it possible to get an event taget off from the custom icon?

**Comment by KresimirJurkovic** [comment]
> Thanks @darkoromanov
> I modified the code to allow the index of those controls
> The custom toolbar options would be like this.
> 
> ```js
>                   customIcons: [{
>                     icon: '<img src="/static/icons/chart-carpet.png" width="20">',
>                     index: 4,
>                     title: 'tooltip of the icon',
>                     class: 'custom-icon',
>                     click: function () {
>                       console.log("cliked")
>                     }
>                   }]
> ```

Is it possible to get an event taget off from the custom icon?

**Comment by sskanishk** [comment]
> Obrigado @darkoromanov Modifiquei
> o código para permitir o índice desses controles
> As opções personalizadas da barra de ferramentas seriam assim.
> 
>                   customIcons: [ { 
>                     icon : '<img src = "/ static / icons / chart-carpet.png" width = "20">' , 
>                     índice : 4 , 
>                     título : 'dica de ferramenta do ícone' , 
>                     classe : 'custom-icon ' , 
>                     clique em : function  ( )  { 
>                       console . log ( "clicado" ) 
>                     } 
>                   } ]
> Please, how do I replace the icon with another svg? Or another local PNG icon?
> 
> This did not work:
> 
>        zoom: `<img src={require('./icons-chart/teste.png')} width="20">`
>        zoom: '<img src={require(`./icons-chart/teste.png`)} width="20">',
>        zoom: '<img src="./icons-chart/teste.png" width="20">',
>        zoom: '<img src="./icons-chart/teste.svg" width="20">',
>        zoom: '<img src=`${varSVG}` width="20">',
> I tried some other ways and none of them rendered.
> 
> To test, I added the two icons (svg and png) to a  tag to make sure they are rendering outside the lib.

Did you get the solution ?

**Comment by yahya-abou-imran** [comment]
I’m experiencing the same problem… Nothing is rendered, even if you double check that the path to the static image is right.

**Comment by lhllyra** [comment]
> > Obrigado @darkoromanov Modifiquei
> > o código para permitir o índice desses controles
> > As opções personalizadas da barra de ferramentas seriam assim.
> > ```
> >               customIcons: [ { 
> >                 icon : '<img src = "/ static / icons / chart-carpet.png" width = "20">' , 
> >                 índice : 4 , 
> >                 título : 'dica de ferramenta do ícone' , 
> >                 classe : 'custom-icon ' , 
> >                 clique em : function  ( )  { 
> >                   console . log ( "clicado" ) 
> >                 } 
> >               } ]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Please, how do I replace the icon with another svg? Or another local PNG icon?
> > This did not work:
> > ```
> >    zoom: `<img src={require('./icons-chart/teste.png')} width="20">`
> >    zoom: '<img src={require(`./icons-chart/teste.png`)} width="20">',
> >    zoom: '<img src="./icons-chart/teste.png" width="20">',
> >    zoom: '<img src="./icons-chart/teste.svg" width="20">',
> >    zoom: '<img src=`${varSVG}` width="20">',
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I tried some other ways and none of them rendered.
> > To test, I added the two icons (svg and png) to a  tag to make sure they are rendering outside the lib.
> 
> Did you get the solution ?

Solved in #481 , instantiated my custom icon as an svg tag through an string literal constant and assigned it to the download icon. Check my comment over there for more details and images of the solution.

Happy coding!
